src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/InstantiateArgumentExceptionsCorrectly.cs

```diff
@@ -112,39 +115,56 @@ public override void Initialize(AnalysisContext analysisContext)
                         continue;
                     }
 
-                    CheckArgument(owningSymbol, creation, argument.Parameter, value, context);
+                    diagnostic = CheckArgument(owningSymbol, creation, argument.Parameter, value, context);
+                    if (diagnostic != null && !diagnostic.Properties.IsEmpty)
+                    {
+                        break;
+                    }
+                }
+
+                if (diagnostic != null)
+                {
+                    context.ReportDiagnostic(diagnostic);
                 }
             }
         }
-        private static void CheckArgument(
+
+        private static bool MatchesConfiguredVisibility(ISymbol owningSymbol, OperationAnalysisContext context)
+        {
+            return owningSymbol.MatchesConfiguredVisibility(context.Options, RuleIncorrectParameterName,
+                context.Compilation, context.CancellationToken, defaultRequiredVisibility: SymbolVisibilityGroup.All);
```

Comment:
I am wondering why the api surface configuration missing `protected` visibility is it fall into `public` or ...? CC @mavasani

Reply:
Do you want the rule to run on all methods by default or just public API surface? Your current approach will do the former, but I think you want the latter? If so, you should remove the explicit argument `defaultRequiredVisibility`, as the default argument for this parameter ensures only public API surface is analyzed.

Reply:
I think it better to leave it as `All` by default as we can fine tune it using `api_surface` option, my plan was to allow all by default and set public/protected option for runtime repo, but protected option is missing which is why am asking about that.

Reply:
You should also be able to inline this method...

Reply:
> my plan was to allow all by default 

Ah, sorry, I misunderstood your desire. Your current code if fine then .

> protected option is missing which is why am asking about that

`SymbolVisibility` is actually computed by walking the containing symbol chain and computing overall visibility. So, a protected method within an internal type will have `SymbolVisibility.Internal`, while protected member within a public type will have `SymbolVisibility.Public`. It is basically effective member visibility.

See https://github.com/dotnet/roslyn-analyzers/blob/eb943221428db2173668b07204759cd64db134e5/src/Utilities/Compiler/Extensions/ISymbolExtensions.cs#L249-L294

Reply:
Got it, thank you!